### PR TITLE
[Ansible/en] Spelling / Grammar

### PR DESCRIPTION
**Pull request by PatMyron**
- [x] I solemnly swear that this is all original content of which I am the original author
- [x] Pull request title is prepended with `[language/lang-code]`
- [x] Pull request touches only one file (or a set of logically related files with similar changes made)
- [x] Content changes are aimed at *intermediate to experienced programmers* (this is a poor format for explaining fundamental programming concepts)
- [x] If you've changed any part of the YAML Frontmatter, make sure it is formatted according to [CONTRIBUTING.md](https://github.com/adambard/learnxinyminutes-docs/blob/master/CONTRIBUTING.markdown)
  - [x] Yes, I have double-checked quotes and field names!

### CODE REVIEW

File: ansible.html.markdown [hunk 1]
```diff
@@ -182,7 +182,7 @@ _Note: Example playbook is explained in the next chapter: 'Roles'
 ### More on ansible concept
 
 ### Inventory
-Inventory is a set of an objects or hosts, against which we are executing our playbooks or single tasks via shell commands
+Inventory is a set of objects or hosts, against which we are executing our playbooks or single tasks via shell commands
 For this few minutes, lets asume that we are using default ansible inventory (which in Debian based system is placed in /etc/ansible/hosts)
```
Comment (Menelion) [inline ansible.html.markdown]: For these few minutes, let's assume that we use the default Ansible inventory

File: ansible.html.markdown [hunk 2]
```diff
@@ -207,7 +207,7 @@ some_other_group
 A concept called `role` was introduced for parts of the code (playbooks) that should be reusable.
 
 **Role** is a structured way to manage your set of tasks, variables, handlers, default settings, and way more (meta, files, templates).
-Role allows to reuse the same parts of code in multiple plybooks (you can parametrize the role 'further' during it's execution).
+Role allows to reuse the same parts of code in multiple playbooks (you can parametrize the role 'further' during it's execution).
```
Comment (Menelion) [inline ansible.html.markdown]: its execution (it's == it is)

File: ansible.html.markdown [hunk 3]
```diff
@@ -207,7 +207,7 @@ some_other_group
...
 It is a great way to introduce `object oriented` management for your applications.
 
 Role can be included in your playbook (executed via your playbook).
```
Comment (Menelion) [inline ansible.html.markdown]: "roles" or "A role"

File: ansible.html.markdown [hunk 4]
```diff
@@ -237,7 +237,7 @@ This example install ansible in `virtualenv` so it is independend from a system.
 We are going to use repository with examples: https://github.com/sirkubax/ansible-for-learnXinYminutes
```
Comment (Menelion) [inline ansible.html.markdown]: a repository

File: ansible.html.markdown [hunk 5]
```diff
@@ -268,8 +268,8 @@ roles/
 ```
 
 #### Role Handlers
-Handlers are a tasks that can be triggered (notified) during execution of a playbook, but they itself execute at the very end of a playbook.
-It is a best way to restart a service, check if application port is active (successfull deployment criteria), etc.
+Handlers are tasks that can be triggered (notified) during execution of a playbook, but they itself execute at the very end of a playbook.
```
Comment (Menelion) [inline ansible.html.markdown]: they themselves?

File: ansible.html.markdown [hunk 6]
```diff
@@ -268,8 +268,8 @@ roles/
...
+It is a best way to restart a service, check if application port is active (successful deployment criteria), etc.
```
Comment (Menelion) [inline ansible.html.markdown]: an/the application port (depends on context)

File: ansible.html.markdown [hunk 7]
```diff
@@ -268,8 +268,8 @@ roles/
...
 
 Please get familiar how you can use role in simple_apache_role example
```
Comment (Menelion) [inline ansible.html.markdown]: roles, or a role, or \`role\`

File: ansible.html.markdown [hunk 8]
```diff
@@ -321,7 +321,7 @@ ansible -m shell -a 'echo "{{ my_variable }}"' -e 'my_variable="{{ lookup("pipe"
 ### Register and Conditional 
 
 #### Register
-Another way to dynamicaly generate the variable content is a `register` command.
+Another way to dynamically generate the variable content is a `register` command.
```
Comment (Menelion) [inline ansible.html.markdown]: the \`register\` command

File: ansible.html.markdown [hunk 9]
```diff
@@ -321,7 +321,7 @@ ansible -m shell -a 'echo "{{ my_variable }}"' -e 'my_variable="{{ lookup("pipe"
...
 `Register` is also useful to store an output of a task, and use it's value as a logic
```
Comment (Menelion) [inline ansible.html.markdown]: its value

File: ansible.html.markdown [hunk 10]
```diff
@@ -321,7 +321,7 @@ ansible -m shell -a 'echo "{{ my_variable }}"' -e 'my_variable="{{ lookup("pipe"
 ### Register and Conditional 
 
 #### Register
-Another way to dynamicaly generate the variable content is a `register` command.
+Another way to dynamically generate the variable content is a `register` command.
 `Register` is also useful to store an output of a task, and use it's value as a logic 
 for execution further tasks.
```
Comment (Menelion) [inline ansible.html.markdown]: for execution of further tasks or for further tasks execution.

File: ansible.html.markdown [hunk 11]
```diff
@@ -398,7 +398,7 @@ You can limit an execution of your tasks to defined hosts
 
 ### Templates
 
-Template is a powerfull way to deliver some (partially) dynamic content. Ansible uses **Jinja2** langueage to describe the template.
+Template is a powerful way to deliver some (partially) dynamic content. Ansible uses **Jinja2** langueage to describe the template.
```
Comment (Menelion) [inline ansible.html.markdown]: Templates are (or a template is); language

File: ansible.html.markdown [hunk 12]
```diff
@@ -409,7 +409,7 @@ Some static content
     this line item is {{ item }}
 {% endfor %}
 ```
-Jinja may have some limitations, but it is a powerfull tool that you might like.
+Jinja may have some limitations, but it is a powerful tool that you might like.
 
 Please examine this simple example that install apache2 and generate index.html from the template
```
Comment (Menelion) [inline ansible.html.markdown]: installs and generates

File: ansible.html.markdown [hunk 13]
```diff
@@ -442,7 +442,7 @@ In fact - jinja is used to template parts of the playbooks too
 ```
 
 #### Jinja2 filters
-Junja is powerfull. It has built-in many usefull functions.
+Jinja is powerful. It has built-in many useful functions.
```
Comment (Menelion) [inline ansible.html.markdown]: It has many built-in useful functions? Or even better: It has many useful built-in functions.

File: ansible.html.markdown [hunk 14]
```diff
@@ -452,7 +452,7 @@ Junja is powerfull. It has built-in many usefull functions.
 [Read More](http://docs.ansible.com/ansible/latest/playbooks_filters.html)
 
 ### ansible-vault
-To maintain **ifrastructure as a code** you need to store secrets. 
+To maintain **infrastructure as code** you need to store secrets. 
   Ansible provides a way to encrypt the confidential files so you can store it in the repository, yet the files are decrypted in-fly during ansible execution.
```
Comment (Menelion) [inline ansible.html.markdown]: Ansible provides a way to encrypt confidential files so you can store them in the repository, yet the files are decrypted on-the-fly during ansible execution.

File: ansible.html.markdown [hunk 15]
```diff
@@ -452,7 +452,7 @@ Junja is powerfull. It has built-in many usefull functions.
...
 
 The best way to use the **ansible-vault** is to store the secret in some secure location, and configure ansible to use during runtime.
```
Comment (Menelion) [inline ansible.html.markdown]: to use it

File: ansible.html.markdown [hunk 16]
```diff
@@ -490,7 +490,7 @@ You might like to know, that you can build your inventory dynamically.
 
 (For Ansible) inventory is just a JSON with proper structure - if you can deliver that to ansible - anything is possible.
```
Comment (Menelion) [inline ansible.html.markdown]: an inventory is just JSON?

File: ansible.html.markdown [hunk 17]
```diff
@@ -490,7 +490,7 @@ You might like to know, that you can build your inventory dynamically.
...
 
-You do not need to invent the wheel - there are plenty ready to use inventory script for most popular Cloud provicers and a lot of in-house popular usecaseses.
+You do not need to invent the wheel - there are plenty ready to use inventory script for most popular Cloud provicers and a lot of in-house popular usecases.
```
Comment (Menelion) [inline ansible.html.markdown]: You do not need to reinvent the wheel - there are plenty of ready to use inventory scripts for most popular Cloud providers and a lot of in-house popular use cases.

File: ansible.html.markdown [hunk 18]
```diff
@@ -514,8 +514,8 @@ callback_whitelist = profile_tasks
 ```
 
 ### facts-cache and ansible-cmdb
-You can pool some infrmations of you environment from another hosts.
-If the informations does not change - you may consider using a facts_cache to speed things up.
+You can pool some information of you environment from another hosts.
```
Comment (Menelion) [inline ansible.html.markdown]: You can pool (or pull?) some information about your environment from other hosts.

File: ansible.html.markdown [hunk 19]
```diff
@@ -533,7 +533,7 @@ fact_caching_timeout = 86400
 I like to use `jsonfile` as my backend. It allows to use another project
 `ansible-cmdb` [(project on github)](https://github.com/fboender/ansible-cmdb) that generates a HTML page of your inventory resources. A nice 'free' addition!
 
-### debugging ansible [chapter in progres]
+### debugging ansible [chapter in progress]
 When your job fails - it is good to be effective with debugging.
 
 1. Increase verbosiy by using multiple -v  **[ -vvvvv]**
```
Comment (Menelion) [inline ansible.html.markdown]: verbosity

File: ansible.html.markdown [hunk 20]
```diff
@@ -546,7 +546,7 @@ When your job fails - it is good to be effective with debugging.
 
 ### Infrastructure as a code
```
Comment (Menelion) [inline ansible.html.markdown]: as code?

File: ansible.html.markdown [hunk 21]
```diff
@@ -546,7 +546,7 @@ When your job fails - it is good to be effective with debugging.
...
 You already know, that ansible-vault allow you to store your confidential data along with your code (in repository). You can go further - and define your ansible installation and configuration as-a-code.
```
Comment (Menelion) [inline ansible.html.markdown]: allows

File: ansible.html.markdown [hunk 22]
```diff
@@ -546,7 +546,7 @@ When your job fails - it is good to be effective with debugging.
 
 ### Infrastructure as a code
 You already know, that ansible-vault allow you to store your confidential data along with your code (in repository). You can go further - and define your ansible installation and configuration as-a-code. 
-See `environment.sh` to learn how to install the ansible itself inside a `virtualenv` that is not attached to your operating system (can be changed by non-privilages user), and as additional benefit - upgrading version of ansible is as easy as installing new version in new virtualenv. What is more, you can have multiple versions of Ansible present in the same time. This is very helpfull!
+See `environment.sh` to learn how to install the ansible itself inside a `virtualenv` that is not attached to your operating system (can be changed by non-privilages user), and as additional benefit - upgrading version of ansible is as easy as installing new version in new virtualenv. What is more, you can have multiple versions of Ansible present in the same time. This is very helpful!
```
Comment (Menelion) [inline ansible.html.markdown]: non-privileged user; at the same time

File: ansible.html.markdown [hunk 23]
```diff
@@ -583,7 +583,7 @@ Note: You may like to execute Ansible with `--ask-sudo-pass` or add the user to
 Always make sure that your playbook can executes in 'dry run' mode (--check), and it's execution is not declaring 'Changed' objects.
```
Comment (Menelion) [inline ansible.html.markdown]: "can execute" or "executes"; its

File: ansible.html.markdown [hunk 24]
```diff
@@ -622,9 +622,9 @@ You can register the output (stdout), rc (return code), stderr of a task with th
 
 
 ## Introduction
-Ansible is (one of the many) orchestration tools. It allows you to controll your environment (infrastructure and a code) and automate the manual tasks.
+Ansible is (one of the many) orchestration tools. It allows you to control your environment (infrastructure and a code) and automate the manual tasks.
```
Comment (Menelion) [inline ansible.html.markdown]: and code

File: ansible.html.markdown [hunk 25]
```diff
@@ -622,9 +622,9 @@ You can register the output (stdout), rc (return code), stderr of a task with th
...
 'You can think as simple as writing in bash with python API 
-Of course the rabit hole is way deeper.'
+Of course the rabbit hole is way deeper.'
 
 Ansible have great integration with multiple operating systems (even Windows) and some hardware (switches, Firewalls, etc). It has multiple tools that integrate with the could providers. Almost every worth-notice cloud provider is present in the ecosystem (AWS, Azure, Google, DigitalOcean, OVH, etc...)
```
Comment (Menelion) [inline ansible.html.markdown]: Ansible has; cloud (not could); noteworthy

File: ansible.html.markdown [hunk 26]
```diff
@@ -648,12 +648,12 @@ It is very-very-very easy to start. If you are familiar with ssh concept - you a
 I do provide a training services - I'm able to teach a production-ready person - in 8 hours (1 training day)! It covers all needed to work aspects! No other tool can match this ease of use!
```
Comment (Menelion) [inline ansible.html.markdown]: provide training services

**Comment by PatMyron** [comment]
@Menelion Sorry, I already deleted that branch and the fork itself. Could you merge these changes in and then I can open another pull request with the additional changes?

**Comment by Menelion** [comment]
@PatMyron done.

**Pull request merged**